sdk/src/azure/core/az_credential_client_secret.c

```diff
@@ -125,7 +114,7 @@ AZ_NODISCARD az_result az_credential_client_secret_init_with_authority(
         .client_id = client_id,
         .client_secret = client_secret,
         .scopes = { 0 },
-        .authority = authority,
+        .authority = az_span_size(authority) > 0 ? authority : azure_ad_global_authority,
```

Comment:
Nice!

Reply:
Exactly, this is the context when az_aad_global_authority becomes the default.